db/sql/sql.go

```diff
@@ -563,12 +588,18 @@ func (tx *Transaction) logQuery(sql string, args ...interface{}) {
 	log.Debug("[%p] Executing SQL query '%s'", tx.transaction, query)
 }
 
+func (tx *Transaction) measureTime(timeStarted time.Time, schemaId, action string) {
+	metrics.UpdateTimer(timeStarted, "tx.%s.%s", schemaId, action)
+}
+
 func (tx *Transaction) Exec(sql string, args ...interface{}) error {
 	return tx.ExecContext(context.Background(), sql, args...)
 }
 
 // Exec executes sql in transaction
 func (tx *Transaction) ExecContext(ctx context.Context, sql string, args ...interface{}) error {
```

Comment:
`ExecContext` may be invoked from other interface methods, resulting in not reliable data. I'll fix it (later) with a small refactoring: public methods shouldn't be invoked from other public methods, so I'll extract the implementation of `ExecContext` into a private `exec` method. The metrics will be gathered only in on the interface level, not in the helper.

Reply:
done